java/vector/src/main/java/org/apache/arrow/vector/NullableBitVector.java

```diff
@@ -163,7 +163,7 @@ private ArrowBuf splitAndTransferBuffer(int startIndex, int length,
             if (destBuffer != null) {
                destBuffer.release();
             }
-            destBuffer = destBuffer.slice(firstByteSource, byteSizeTarget);
```

Comment:
Are there unit tests for this kind of stuff?

Reply:
Yes, we have unit tests for splitAndTransfer of bit vector (validity buffer). Also we have unit tests  for split and transfer of other vectors (nullable fixed, var width and complex) and we exercise all the conditions.